python/tank/bootstrap/resolver.py

```diff
@@ -25,139 +25,102 @@
-        :return: Configuration instance
+        :param config_descriptor: descriptor dict or string
+        :param sg_connection: Shotgun API instance
+        :return: :class:`Configuration` instance
         """
-        log.debug(
-            "%s resolving a configuration for project %s, "
-            "pipeline config %s, engine %s" % (
-                self,
-                project_id,
-                pipeline_config_name,
-                engine_name
-            )
-        )
+        log.debug("%s resolving configuration for descriptor %s" % (self, config_descriptor))
 
-        # fall back on base
-        if base_config_descriptor is None:
+        if config_descriptor is None:
             raise TankBootstrapError(
-                "No base configuration specified and no pipeline "
-                "configuration exists in Shotgun for the given project. "
-                "Cannot create a configuration object.")
+                "No config descriptor specified - Cannot create a configuration object."
+            )
 
+        # note how we currently always prefer latest as part of the resolve.
+        # later on, it will be possible to specify an update policy as part of
+        # the descriptor system, allowing a user to specify what latest means -
+        # does it actually mean that the version is frozen to a particular version,
+        # the latest release on a conservative release track, the latest alpha etc.
+        #
+        # for installations bundled with manual descriptors, latest currently
+        # means the same version that the descriptor is pointing at, so for
+        # these installations, version numbers are effectively fixed.
         cfg_descriptor = create_descriptor(
-            self._sg_connection,
+            sg_connection,
             Descriptor.CONFIG,
-            base_config_descriptor,
+            config_descriptor,
             fallback_roots=self._bundle_cache_fallback_paths,
-            resolve_latest=get_latest_config
+            resolve_latest=True
         )
 
         log.debug("Configuration resolved to %r." % cfg_descriptor)
 
         # first get the cache root
         cache_root = LocalFileStorageManager.get_configuration_root(
-            self._sg_connection.base_url,
-            project_id,
+            sg_connection.base_url,
+            self._project_id,
             None,  # pipeline config id
             LocalFileStorageManager.CACHE
         )
 
+        # resolve the config location both based on entry point and current engine.
+        #
+        # Example: ~/Library/Caches/Shotgun/mysitename/site/cfg.tk-rv.review
+        #
+        config_folder = "cfg.%s" % filesystem.create_valid_filename(self._engine_name)
+
+        if self._entry_point:
+            # append the entry point
+            config_folder = "%s.%s" % (
+                config_folder,
+                filesystem.create_valid_filename(self._entry_point)
+            )
```

Comment:
With 0.18, paths are reshuffled and everything is already compressed. I think things are already pretty compressed, but this is likely to change soon with #36909, which I think will mean we can shorten it further.